Домашняя работа спринт 12

KozlovskiyD opened the pull request.

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12

faritowich commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -17,15 +20,26 @@ import retrofit2.Retrofit
 
 
+const val DATA = "data"
+const val KEY = "key"
```
лучше не объявлять вне класса в файле. Лучше тогда уж объявить в companion object и сделать private, если использовать планируется только в данном классе

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -111,5 +113,40 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/top_24"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"/>
```
Поиск песен по сети и история поиска используют одинаковые лейауты, лучше в таких случаях объединять их в один адаптер и в зависимости от режима истории/режима поиска показывать/скрывать надпись "Вы искали" и кнопку "Очистить".

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12

faritowich commented on app/src/main/java/com/practicum/playlistmaker3/SearchActivity.kt:
```diff
@@ -103,28 +147,48 @@ class SearchActivity : AppCompatActivity() {
         rvTrackList.visibility = View.GONE
     }
+
+    private fun loadHistory() {
```
Этому методу тоже место в SearchHistory

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12

Dmitr80 committed:
Init commit
Toast all
Домашная работа часть 3.5
Домашная работа спринт 9.1
Домашняя работа спринт 10.1
домашняя работа спринт 12
Домашняя работа спринт13
